### PR TITLE
feat: main functions

### PR DESCRIPTION
**Pull request by Nekotoxin**
@leeqvip Is there anything that is not as expected? If not, I will continue to finish the rest of the project.

ps: Preliminary completion of request_enforcer

### CODE REVIEW

File: dauthz/core.py [hunk 1]
```diff
@@ -1,13 +1,114 @@
...
+def init_enforcers():
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT app, name applied FROM django_migrations
+                WHERE app = 'casbin_adapter' AND name = '0001_initial';
+                """
+            )
+            row = cursor.fetchone()
```
Comment (leeqvip) [inline dauthz/core.py]: There is no need to decide whether to load the enforcer based on migration. Not sure which adapter is used, if the adapter does not exist, just throw an exception when importing the adapter

File: dauthz/settings.py [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
-            'CONNECTION' : '',
-        },
+            # Casbin adapter .
+            'ADAPTER': 'casbin_adapter.adapter.Adapter',
+            # Database setting.
+            'STORAGE': {
+                # Available Settings: "database", "file"
+                'STORAGE_TYPE': 'database',
+                # Database connection for following tables.
+                'DATABASE_CONNECTION': '',
```
Comment (leeqvip) [inline dauthz/settings.py]: The adapter configuration is like this, the name indicates the import path of the adapter, and the option is the configuration of the adapter, which varies with the adapter.  `watcher` can also be similar.
```
        'ADAPTER': {
            'NAME': 'casbin_adapter.adapter.Adapter',
            'OPTION_1': '',
            'OPTION_2': '',
            # ....
        }
```

**Review by leeqvip — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Nekotoxin plz modify

**Comment by Nekotoxin** [comment]
> @Nekotoxin plz modify

ok. Preliminary completion of request_middleware and enforcer_middleware.
based on: https://github.com/pycasbin/django-orm-adapter/pull/16
@leeqvip plz review.

File: dauthz/core.py [hunk 3]
```diff
@@ -1,13 +1,46 @@
...
+        watcher = enforcer_conf.get("WATCHER")
+        if watcher:
+            self.set_watcher(watcher)
+        role_manager = enforcer_conf.get("ROLE_MANAGER")
+        if role_manager:
+            self.set_role_manager(role_manager)
+        self.logger.debug("Casbin enforcer initialised")
 
 
-enforcer = ProxyEnforcer()
```
Comment (leeqvip) [inline dauthz/core.py:13]: There should be a default enforcer here.

File: dauthz/decorators/enforcer_middleware.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
-from dauthz.middlewares.enforcer import EnforcerMiddleward
 from django.utils.decorators import decorator_from_middleware
 
-enforcer_middleward = decorator_from_middleware(EnforcerMiddleward)
\ No newline at end of file
+from dauthz.middlewares.enforcer import EnforcerMiddleware
+
+enforcer_middleward = decorator_from_middleware(EnforcerMiddleware)
```
Comment (leeqvip) [inline dauthz/decorators/enforcer_middleware.py]: `middleware`, not `middleward`

**Review by leeqvip — CHANGES_REQUESTED** [review]

File: dauthz/core.py (hunk 3 above)
Comment (Nekotoxin) [reply dauthz/core.py:13]: image
image
The default enforcer is specified by passing the DEFAULT field, but the user can also modify the enforcer as needed

File: dauthz/core.py [hunk 5]
```diff
@@ -1,13 +1,46 @@
...
+enforcers = {}
```
Comment (leeqvip) [inline dauthz/core.py:42]: I think this provides an entry for external calls like this:

```python
dauthz.core.enforcer.enforce(...)
dauthz.core.enforcer.get_policy(...)
...
```

**Review by leeqvip — CHANGES_REQUESTED** [review]

File: dauthz/core.py (hunk 5 above)
Comment (Nekotoxin) [reply dauthz/core.py:42]: changed. add default enforcer.
image
set the default enforcer in the conf file like laravel-authz

**Review by leeqvip — APPROVED** [review]

**Pull request merged**